### PR TITLE
[14.0] ADD account_receipt_sale

### PR DESCRIPTION
**Pull request by eLBati**
Depends on https://github.com/OCA/account-invoicing/pull/1260 and https://github.com/OCA/account-invoicing/pull/1427

### CODE REVIEW

File: account_receipt_sale/models/account_fiscal_position.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            # Fall back to fiscal positions without company
+            receipt_fiscal_pos = self.search(
+                [
+                    ('company_id', '=', False),
+                    ('receipts', '=', True),
+                ],
+                limit=1
+            )
+
+        return receipt_fiscal_pos
```
Comment (primes2h) [inline account_receipt_sale/models/account_fiscal_position.py]: Visto quanto indicato [qui](https://github.com/OCA/l10n-italy/issues/2722#issuecomment-1272490383) direi che questa parte non serve.
O quantomeno non dovrebbe stare in questo modulo, se vogliamo che lo stesso corrisponda allo scopo di `l10n_it_corrispettivi_sale` della 12.0.

File: account_receipt_sale/models/partner.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            # fiscal position only if there is none
+            if not self.property_account_position_id:
+                company = self.company_id or self.env.company
+                self.property_account_position_id = \
+                    self.env['account.fiscal.position'] \
+                        .get_receipts_fiscal_pos(company)
+        else:
+            # Unset the fiscal position only if it was receipts
+            if self.property_account_position_id.receipts:
+                self.property_account_position_id = False
```
Comment (primes2h) [inline account_receipt_sale/models/partner.py]: Vedi sopra.

File: account_receipt_sale/views/account_fiscal_position_views.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        <field name="model">account.fiscal.position</field>
+        <field name="type">form</field>
+        <field name="inherit_id" ref="account.view_account_position_form"/>
+        <field name="arch" type="xml">
+            <field name="company_id" position="after">
+                <field name="receipts"/>
+            </field>
+        </field>
+    </record>
+</odoo>
```
Comment (primes2h) [inline account_receipt_sale/views/account_fiscal_position_views.xml]: Vedi sopra.

File: account_receipt_sale/views/partner_views.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="account.view_partner_property_form"/>
+        <field name="groups_id" eval="[(4, ref('account.group_account_manager'))]"/>
+        <field name="arch" type="xml">
+            <group name="fiscal_information" position="inside">
+                <field name="use_receipts"/>
+            </group>
+        </field>
+    </record>
+</odoo>
```
Comment (primes2h) [inline account_receipt_sale/views/partner_views.xml]: Vedi sopra.

**Review by primes2h — COMMENTED** [review]
Vedi anche https://github.com/OCA/l10n-italy/issues/2722#issuecomment-1387195027

**Comment by primes2h** [comment]
I'm sorry for comments in italian language, I tought i was in the italian repo (l10n-italy). :sweat_smile:

File: account_receipt_sale/models/account_fiscal_position.py (hunk 1 above)
Comment (eLBati) [reply account_receipt_sale/models/account_fiscal_position.py]: The purpose of `account_receipt_sale` is essentially in its [DESCRIPTION.rst](https://github.com/OCA/account-invoicing/blob/828fa64a7b74baf0da56bb75aede0c28d05afcda/account_receipt_sale/readme/DESCRIPTION.rst)
so it includes parts of `l10n_it_corrispettivi` and `l10n_it_corrispettivi_sale`

**Comment by rafaelbn** [comment]
@eLBati hello! Dependency merged!

**Comment by tafaRU** [comment]
@eLBati could you please rebase in order to rebuild runboat?

**Review by SerM11 — COMMENTED** [review]
Tested :

From back end create correctly the receipts when the flag on partner is set.
Also from e-commerce is possible to create a receipt for the SO linked to a partner with receipt flag active 

On receipts don't see any information about the journal used

I don't see any menu in invoicing module for access to the receipts list (but from the single partner is possible to access their ones)

the flag "group_show_sale_receipts" (under configuration / settings / invoice --> Sale Receipt Enterprise) is a enterprise feature and require an upgrade to enterprise environment in v14 (in v16 no more).

**Comment by SerM11** [comment]
@eLBati any news? thanks

**Comment by SirTakobi** [comment]
> I don't see any menu in invoicing module for access to the receipts list (but from the single partner is possible to access their ones)

Here is the menu:
image
you need the group `account.group_sale_receipts` in order to see it, have a look at https://github.com/odoo/odoo/blob/960e61e7451ef8dacd3aaa70bb148240762ab461/addons/account/views/account_menuitem.xml#L13

**Review by SerM11 — APPROVED** [review]
@SirTakobi thank you, can you please add this step to the readme / instructions of the module?

I did other functional test and looks good to me.
:)

**Comment by SirTakobi** [comment]
> can you please add this step to the readme / instructions of the module?

cc @eLBati

**Review by Sashimi22 — APPROVED** [review]
Functional test ok

**Comment by eLBati** [comment]
> can you please add this step to the readme / instructions of the module?

Done!

**Pull request merged**